### PR TITLE
Lint for using suite.T().Run instead of testify's suite.Run

### PR DESCRIPTION
**Pull request by ashmrtn**
Includes fixups for calls that need updated

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [x] :robot: Test
- [x] :computer: CI/Deployment
- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #2373

#### Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/pkg/selectors/exchange_test.go [hunk 1]
```diff
@@ -805,7 +809,7 @@ func (suite *ExchangeSelectorSuite) TestIdPath() {
 		},
 	}
 	for _, test := range table {
-		suite.T().Run(test.cat.String(), func(t *testing.T) {})
+		suite.Run(test.cat.String(), func() {})
```
Comment (ashmrtn) [inline src/pkg/selectors/exchange_test.go:812]: @ryanfkeepers This test isn't actually running anything. Can we file an issue/make a PR to fix that?

Comment (ryanfkeepers) [reply src/pkg/selectors/exchange_test.go:812]: mmm, lets just delete the test.  I'm not even sure what it's supposed to be testing, and I have a suspicion that it's vestigial.

Comment (ashmrtn) [reply src/pkg/selectors/exchange_test.go:812]: see #2665

File: src/.golangci.yml [hunk 2]
```diff
@@ -34,6 +34,9 @@ linters-settings:
       - '\bpath\.(Ext|Base|Dir|Join)'
       # Don't allow the typo m356 to be used in place of m365.
       - '[Mm]356'
+      # Prefer suite.Run(name, func() {}) for subtests as testify has it instead
+      # of suite.T().Run(name, func(t *testing.T) {}).
+      - 'T\(\)\.Run(# prefer testify subtest Run(name, func()) function )?'
```
Comment (ryanfkeepers) [inline src/.golangci.yml:39]: Might want the following, as the current regex doesn't catch t.Run().
```suggestion
      - '(T\(\)|t)\.Run(# prefer testify suite.Run(name, func()) )?'
```

**Review by ryanfkeepers — APPROVED** [review]

**Pull request merged**